client/src/app/components/page/task-detail/task-detail.component.ts

```diff
@@ -86,14 +80,72 @@ export class TaskDetailComponent implements OnInit, OnDestroy {
     modalRef.result.then((result) => {
       if (result) {
         this.ts.deleteTask(this.id).pipe(takeUntil(this.ngUnsubscribe)).subscribe(res => {
-          console.log(res)
           this.router.navigate(['/tasks']);
         })
       }
     });
   }
-  public ngOnDestroy(){
+  public ngOnDestroy() {
     this.ngUnsubscribe.next();
     this.ngUnsubscribe.complete();
   }
+  private getTags() {
+    return this.ts.getTags().subscribe(
+      tags => this.prepareTagsSelect(tags)
+    );
+  }
+
+  private prepareTagsSelect(tags) {
+    tags.forEach((item, index) => {
+      this.allUniqueTags = [...this.allUniqueTags, { id: index, text: item }];
+    });
+  }
+  public selectChanged(event) {
+    if (!this.tags.includes(event.data[0].text) && !this.firstSelect) {
+      this.tags.push(event.data[0].text)
+      // I don't know if the user should be able to choose tags which are already in use
+      //this.allUniqueTags = this.allUniqueTags.filter((_, index) => index != event.value)  
+    }
+    this.firstSelect = false;
+  }
+  public deleteTag(index) {
+    this.tags = this.tags.filter((_, i) => i !== index);
+  }
+  private initialize() {
+    this.id = this.route.snapshot.paramMap.get('id');
+    this.taskForm = this.fb.group({
+      title: [''],
+      description: ['']
+    });
+    if (this.id === 'newTask') {
+      this.updateMode = false;
+      this.tags = [];
+      this.buttonText = 'Add task';
+    }
+    else {
+      this.ts.getTaskById(this.id).pipe(takeUntil(this.ngUnsubscribe)).subscribe(task => {
+        this.task = task;
+        this.tags = task.tags;
+        this.taskForm.controls['title'].setValue(this.task.title);
+        this.taskForm.controls['description'].setValue(this.task.description);
+      })
+    }
+    this.inputNewTag = this.fb.group({
+      tag: ['']
+    })
+  }
+  public addNewTag() {
+    const tag = this.inputNewTag.value.tag;
+    if (!this.tags.includes(tag)) {
+      this.tags.push(tag);
+      this.allUniqueTags = [...this.allUniqueTags, tag];
```

Comment:
When I use push the tag doesn't show up in the select2. I read somewhere that's because of how angular detects changes but can't find the article/answer.